### PR TITLE
[graphql/rpc] expose timeout in limits config

### PR DESCRIPTION
**Pull request by oxade**
## Description 

Exposes request timeout setting in config.

## Test Plan 

Existing

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-graphql-rpc/src/extensions/timeout.rs [hunk 1]
```diff
@@ -9,49 +9,32 @@ use std::sync::Arc;
...
-    pub config: TimeoutConfig,
-}
+pub(crate) struct Timeout;
 
 impl ExtensionFactory for Timeout {
     fn create(&self) -> Arc<dyn Extension> {
-        Arc::new(TimeoutExtension {
-            config: self.config,
-        })
+        Arc::new(Timeout {})
```
Comment (amnn) [inline crates/sui-graphql-rpc/src/extensions/timeout.rs:19]: nit: I think the {} are unnecessary here?

```suggestion
        Arc::new(Timeout)
```

File: crates/sui-graphql-rpc/src/server/builder.rs [hunk 2]
```diff
@@ -448,7 +448,7 @@ pub mod tests {
     pub async fn test_query_complexity_metrics_impl() {
         let (connection_config, _cluster) = prep_cluster().await;
 
-        let binding_address: SocketAddr = "0.0.0.0:9184".parse().unwrap();
+        let binding_address: SocketAddr = "0.0.0.0:9185".parse().unwrap();
```
Comment (amnn) [inline crates/sui-graphql-rpc/src/server/builder.rs:451]: Is this change intentional?

**Review by amnn — APPROVED** [review]

File: crates/sui-graphql-rpc/src/server/builder.rs (hunk 2 above)
Comment (oxade) [reply crates/sui-graphql-rpc/src/server/builder.rs:451]: yes, intermittent port clobberting
